### ISSUE TITLE
Issue with reproducing results

### ISSUE THREAD

**Issue by Marshall-mk** [HReynaud/EchoNet-Synthetic#5]
Hi Hadrien,

I was trying to reproduce the results reported in the Echonet-Synthetic paper using the released model checkpoints, following the detailed instructions for the LIDM [here](https://github.com/HReynaud/EchoNet-Synthetic/blob/main/echosyn/lidm/README.md), I synthesized 100,000 images and here is metrics I obtained. 

Echonet-Dynamic Dataset  | FID↓ | IS±std↑ | FID↓ | IS±std↑
-- | -- | -- | -- | --
LIDM **Reported** Results before and after filtering | 17.3 | 2.42±0.02 | 16.4 | 2.37±0.02
LIDM **Reproduced** Results before and after filtering **(ddpm scheduler)** | 79.46 | 1.87±0.0089 | 71.03 | 1.89±0.010
LIDM **Reproduced** Results before and after filtering **(ddim scheduler)** | 90.12 | 1.87±0.0127 | 74.95 | 1.88±0.010

These results do not correlate with what was reported in the paper (first row) and I was wondering what could be wrong. I proceeded to reproduce the results of the LVDM and here are the results I obtained.

Echonet-Dynamic Dataset  | FID↓ | FVD16 ↓ | FVD128 ↓ | IS±std↑
-- | -- | -- | -- | --
LVDM **Reported** results (10,030 samples) | 28.8 | 103.5 | 280.2 | 2.26±0.08
LVDM **Reproduced** results (10,030 samples) (ddpm) | 38.13 | 219.09 | 270.85 | 2.13±0.055
LVDM **Reproduced** results (10,030 samples) (ddim) | 31.96 | 199.13 | 278.99 | 2.13±0.06

Could you shed some light on this please? Thank you in anticipation.

**Comment by HReynaud** [comment]
Hi Muhammad,

Thank you for your interest in this work !

After checking what could have gone wrong, I believe that the image and video model I uploaded to hugging face were not the right ones, both for the image and video model. I will make sure to fix that ASAP, but in the meantime, could you try to sample again with the demo models: https://huggingface.co/spaces/HReynaud/EchoNet-Synthetic/tree/main/resources

The LIDM should be sampled with DDIM and 64 sampling steps.
The LVDM should be sampled with DDPM and 256 sampling steps (although 64 steps should yield good results already)

The reason I think these are not the right models is that the optimization_step value is way too low on the LIDM (100 when it should be in the tens of thousands). For the LVDM the gap is not huge (60k vs 80k) but it could explain some of the discrepancy you observed. See this line: 
https://huggingface.co/HReynaud/EchoNet-Synthetic/blob/main/lidm_dynamic/config.json#L37

Please let me know the results you get with the demo models, I am currently unable to perform this test myself.

**Comment by Marshall-mk** [comment]
Hi Hadrien,

Thank you for the detailed response and guidance!

I haven’t tested the demo models yet but will do so shortly following your suggestions. I’ll update you with the results as soon as I have them.

**Comment by Marshall-mk** [comment]
Hi Hadrien,
I just got the results for the demo models and here they are:


Table.1 LIDM Results on Echonet-dynamic-100k
Echonet-Dynamic Dataset  | FID↓ | IS±std↑ | FID↓ | IS±std↑
-- | -- | -- | -- | --
LIDM Reported Results before and after filtering | 17.3 | 2.42±0.02 | 16.4 | 2.37±0.02
LIDM Reproduced Results before and after filtering (ddim) | 17.3 | 2.41±0.017 | 18.6 | 2.34±0.015

The LIDM results is consistent and it makes sense that the FID increases a little after filtering out extremely similar samples to the training set. 

For the LVDM, I have these results:


Table:2 LVDM Results on Echonet-dynamic dataset

Echonet-Dynamic Dataset | FID↓ | FVD16 ↓ | FVD128 ↓ | IS±std↑ | R2↑ | MAE↓ | RSME↓
-- | -- | -- | -- | -- | -- | -- | --
LVDM Reported results  | 28.8 | 103.5 | 280.2 | 2.26±0.08 | 0.75 | 4.55 | 6.10
LVDM Reproduced results  | 37.36 | 127.75 | 319.02 | 2.18±0.09 | 0.521 | 6.22 | 8.46

The regression metrics were obtained after relabelling with the regression model trained on real data.

Table:3 Comparison for LVEF regression models performance on Real Data
Models | R2↑ | MAE↓ | RSME↓
-- | -- | -- | --
Reported | 0.81 | 3.98 | 5.29
Reproduced | 0.817 | 3.96 | 5.23
